docs/comm_protocol.md

```diff
@@ -7,19 +7,22 @@ For implementation on the STM32 side, please refer to the example [here](https:/
 
 For implementation on the Jetson side, please refer to the main communicator [here](../Communication/communicator.py).
 
-# Protocol from Jetson to STM32
+# Protocol between Jetson and STM32
+
+Note that the communication should always be single-sided (Otherwise we are reinventing TCP). A response should never **directly** depend on any data in previous packets. Example: send back `rel_pitch += 1` is an undefined behavior.
 
 ## Packet Struct
 
-| Name          | Content                   | Size    |
-| ------------- | ------------------------- | ------- |
-| HEADER        | 2 ASCII char              | 2 bytes |
-| SEQ_NUM       | uint32_t counter          | 4 bytes |
-| REL_YAW       | int32_t discretized float | 4 bytes |
-| REL_PITCH     | int32_t discretized float | 4 bytes |
-| CRC_CHECKSUM  | uint8_t checksum          | 1 bytes |
-| PACK_END      | 2 ASCII char              | 2 bytes |
-| TOTAL         |                           | 17 bytes|
+| Name         | Content                            | Size          | Offset         |
+|--------------|------------------------------------|---------------|----------------|
+| HEADER       | fixed 2 ASCII char ('ST')          | 2 bytes       | 0              |
+| SEQ_NUM      | uint16_t counter                   | 2 bytes       | 2              |
+| DATA_LEN     | uint8_t length of the data section | 1 byte        | 4              |
+| CMD_ID       | uint8_t identifier                 | 1 byte        | 5              |
+| DATA         | struct with fixed length           | 1 - 12 bytes  | 6              |
```

Comment:
Is there any particular reason why the size is chosen to be 12 bytes? For example, if I increase this number, will it cause more packet losses? If experiments had been done to find this number, it should be provided as a sidenote.

Reply:
12 bytes is just the longest data struct (ChassisData, 3 floats) currently defined. It can be any size, theoretically.